### PR TITLE
bpo-38871: Fix lib2to3 to be able to use in edge cases of filter

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue38871

### CODE REVIEW

**Comment by vstinner** [comment]
@benjaminp @pablogsal @isidentical: Would you mind to have a look?

File: Misc/NEWS.d/next/Library/2020-01-01-18-44-52.bpo-38871.3EEOLg.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix :mod:`lib2to3` to be able to use in edge cases when handling filter
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Library/2020-01-01-18-44-52.bpo-38871.3EEOLg.rst]: This is what users will se in the changelog so please describe *what* ege cases didn't work and now do :)

**Review by pablogsal — CHANGES_REQUESTED** [review]

**Comment by corona10** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Library/2020-01-01-18-44-52.bpo-38871.3EEOLg.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Updated the logic of :mod:`lib2to3` when handling filter based statement
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Library/2020-01-01-18-44-52.bpo-38871.3EEOLg.rst]: I would suggest the following:

```
Correctly parenthesize filter-based statements that contain lambda expressions in mod:`lib2to3`.
Patch by Dong-hee Na.
```

Comment (corona10) [reply Misc/NEWS.d/next/Library/2020-01-01-18-44-52.bpo-38871.3EEOLg.rst]: Thanks for the suggestion. I've updated it

**Review by pablogsal — APPROVED** [review]

**Comment by vstinner** [comment]
Thanks @corona10 for updating the NEWS entry. I rewrote the commit message.

**Comment by pablogsal** [comment]
Thanks for the PR, @corona10 !

**Pull request merged**